include/alpaka/mem/global/DeviceGlobalCpu.hpp

```diff
@@ -0,0 +1,109 @@
+/* Copyright 2024 Aurora Perego
+ * SPDX-License-Identifier: MPL-2.0
+ */
+
+#pragma once
+
+#include "alpaka/mem/buf/cpu/Copy.hpp"
+#include "alpaka/mem/global/Traits.hpp"
+#include "alpaka/mem/view/ViewPlainPtr.hpp"
+
+#include <type_traits>
+
+// memcpy specialization for device global variables
+namespace alpaka
+{
+
+    namespace detail
+    {
+        template<typename T>
+        struct DevGlobalTrait<TagCpuOmp2Blocks, T>
+        {
+            using Type = detail::DevGlobalImplGeneric<TagCpuOmp2Blocks, T>;
+        };
+
+        template<typename T>
+        struct DevGlobalTrait<TagCpuOmp2Threads, T>
+        {
+            using Type = detail::DevGlobalImplGeneric<TagCpuOmp2Threads, T>;
+        };
+
+        template<typename T>
+        struct DevGlobalTrait<TagCpuSerial, T>
+        {
+            using Type = detail::DevGlobalImplGeneric<TagCpuSerial, T>;
+        };
+
+        template<typename T>
+        struct DevGlobalTrait<TagCpuTbbBlocks, T>
+        {
+            using Type = detail::DevGlobalImplGeneric<TagCpuTbbBlocks, T>;
+        };
+
+        template<typename T>
+        struct DevGlobalTrait<TagCpuThreads, T>
+        {
+            using Type = detail::DevGlobalImplGeneric<TagCpuThreads, T>;
+        };
+    } // namespace detail
+
+    template<typename TAcc, typename TViewSrc, typename TViewDstFwd, typename TQueue>
+    ALPAKA_FN_HOST auto memcpy(
+        TQueue& queue,
+        alpaka::detail::DevGlobalImplGeneric<TAcc, TViewDstFwd>& viewDst,
+        TViewSrc const& viewSrc) -> void
```

Comment:
This approach seems too generic, as in principle it can match also for a CUDA `DevGlobalImplGeneric`.

Also, `TAcc` should be `TTag`.

Reply:
Can you add a constrain that `TAcc` (actually `TTag`) is one of `TagCpuOmp2Blocks`, `TagCpuOmp2Threads`, `TagCpuSerial`, `TagCpuTbbBlocks`, `TagCpuThreads` ?

There may be a smarter way, but at least this should avoid the wrong matches.

Reply:
By the way, IIUC `TViewDstFwd` here is not really a `View`, but the underlying type of the global variable.
If that's the case, could you rename it to `TType` ?